esmvalcore/preprocessor/_regrid_unstructured.py

```diff
@@ -47,3 +59,309 @@ def regridder(
+        udim = cube.coord_dims('latitude')[0]
+        dim_coords_and_dims = [
+            (c, cube.coord_dims(c)[0]) for c in cube.coords(dim_coords=True) if
+            udim not in cube.coord_dims(c)
+        ]
+        dim_coords_and_dims = [
+            (c, d) if d < udim else (c, d + 1) for (c, d) in
+            dim_coords_and_dims
+        ]
+        dim_coords_and_dims.append((self.tgt_coords[0], udim))
+        dim_coords_and_dims.append((self.tgt_coords[1], udim + 1))
+
+        # (2) Include all auxiliary coordinates that do not span unstructured
+        # grid dimension (also make sure to shift all dimensions to the right
+        # of the unstructured grid to the right by 1)
+        old_aux_coords_and_dims = [
+            (c, cube.coord_dims(c)) for c in cube.coords(dim_coords=False) if
+            udim not in cube.coord_dims(c)
+        ]
+        aux_coords_and_dims = []
+        for (aux_coord, dims) in old_aux_coords_and_dims:
+            dims = tuple(d if d < udim else d + 1 for d in dims)
+            aux_coords_and_dims.append((aux_coord, dims))
+
+        # Create new cube
+        regridded_cube = Cube(
+            regridded_data,
+            dim_coords_and_dims=dim_coords_and_dims,
+            aux_coords_and_dims=aux_coords_and_dims,
+        )
+        regridded_cube.metadata = cube.metadata
+
+        return regridded_cube
+
+    def _get_regridded_data(self, cube: Cube) -> np.ndarray | da.Array:
+        """Get regridded data."""
+        udim = cube.coord_dims('latitude')[0]
+
+        # Cube must not be chunked along latitude and longitude dimension
+        rechunk_cube(cube, ['latitude', 'longitude'])
+
+        # Make sure that masked arrays are filled with nan's so they are
+        # handled properly
+        npx = da if cube.has_lazy_data() else np
+        src_data = npx.ma.filled(cube.core_data(), np.nan)
+
+        shape = [s for (d, s) in enumerate(cube.shape) if d != udim]
+        shape.insert(udim, self.tgt_coords[0].core_points().size)
+        shape.insert(udim + 1, self.tgt_coords[1].core_points().size)
+        new_shape = tuple(shape)
+
+        # Perform actual regridding and assign correct mask
+        regridded_data: np.ndarray | da.Array
+        if cube.has_lazy_data():
+            regridded_data = self._regrid_lazy(
+                src_data, udim, self.weights.dtype
+            )
+            # TODO: add limit='128 MiB' to reshape once dask bug is solved
+            # see https://github.com/dask/dask/issues/10603
+            regridded_data = regridded_data.reshape(new_shape)
+        else:
+            regridded_data = self._regrid_eager(src_data, udim)
+            regridded_data = regridded_data.reshape(new_shape)
+        regridded_data = npx.ma.masked_invalid(regridded_data)
+
+        # Ensure correct dtype
+        if regridded_data.dtype != cube.dtype:
+            regridded_data = regridded_data.astype(cube.dtype)
```

Comment:
Is this desirable if the input cube has integer input data?

Reply:
Very good point! No, it's certainly not desired. It would be nice to use the decorator introduced in https://github.com/ESMValGroup/ESMValCore/pull/2393 here, so maybe we can merge this one first?